internal/brokers/withexamples.go

```diff
@@ -27,23 +26,17 @@ func useExampleBrokers() (string, func(), error) {
 		return "", nil, err
 	}
 
-	ctx, cancel := context.WithCancel(context.Background())
-	done := make(chan struct{})
-	go func() {
-		defer close(done)
-		defer os.RemoveAll(cfgPath)
-		defer busCleanup()
-		if err = examplebroker.StartBus(ctx, cfgPath); err != nil {
-			log.Errorf(ctx, "Error starting examplebroker: %v", err)
-		}
-	}()
-
-	// Give some time for the broker to start
-	time.Sleep(time.Second)
+	conn, err := examplebroker.StartBus(cfgPath)
+	if err != nil {
+		os.RemoveAll(cfgPath)
```

Comment:
Now that it’s not in a defer call, we should probably add a warning if we can’t remove the directory (here and below).